src/Bookings.js

```diff
@@ -1,18 +1,47 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import Search from "./Search.js";
-// import SearchResults from "./SearchResults.js";
+import SearchResults from "./SearchResults.js";
 // import FakeBookings from "./data/fakeBookings.json";
+import CustomerProfile from "./CustomerProfile.js";
 
 const Bookings = () => {
+  const [booking, setBooking] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const [customerProfile, setCustomerProfile] = useState([]);
+
+  useEffect(() => {
+    fetch(`https://cyf-react.glitch.me`)
+      .then(res => res.json())
+      .then(data => setBooking(data));
+  }, []);
   const search = searchVal => {
-    console.info("TO DO!", searchVal);
+    // console.info("TO DO!", searchVal);
+    searchVal = searchVal.toLowerCase();
+    const filterd = booking.filter(
+      item =>
+        item.firstName.toLowerCase() === searchVal ||
+        item.surname.toLowerCase() === searchVal
+    );
+    setBooking(filterd);
+  };
+
+  const showProfile = async id => {
+    const res = await fetch(`https://cyf-react.glitch.me/customers/${id}`);
+    if (res.status === 200) {
+      const data = await res.json();
+      setCustomerProfile(data);
+    }
```

Comment:
can you think of a way this request could make use of `useEffect`? for example, if we had some state tracking when the show profile button was clicked, we could use the dependency array to make the request whenever this state changed